### PR TITLE
[IO] Rename Data

### PR DESCRIPTION
**Pull request by lamriaimen**
According to the guidelines

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by alxbilger** [comment]
[ci-build][with-all-tests]

File: Sofa/Component/IO/Mesh/src/sofa/component/io/mesh/MeshSTLLoader.h [hunk 1]
```diff
@@ -48,8 +48,16 @@ class SOFA_COMPONENT_IO_MESH_API MeshSTLLoader : public sofa::core::loader::Mesh
 
 public:
     //Add Data here
-    Data <unsigned int> _headerSize; ///< Size of the header binary file (just before the number of facet).
-    Data <bool> _forceBinary; ///< Force reading in binary mode. Even in first keyword of the file is solid.
+    SOFA_ATTRIBUTE_DEPRECATED__RENAME_DATA()
+    Data <bool> forceBinary;
```
Comment (alxbilger) [inline Sofa/Component/IO/Mesh/src/sofa/component/io/mesh/MeshSTLLoader.h]:
```suggestion
    Data <bool> _forceBinary;
```
It previously started with an underscore

File: Sofa/Component/IO/Mesh/src/sofa/component/io/mesh/MeshSTLLoader.h [hunk 2]
```diff
@@ -48,8 +48,16 @@ class SOFA_COMPONENT_IO_MESH_API MeshSTLLoader : public sofa::core::loader::Mesh
...
+
+    SOFA_ATTRIBUTE_DEPRECATED__RENAME_DATA()
+    Data <unsigned int> headerSize;
```
Comment (alxbilger) [inline Sofa/Component/IO/Mesh/src/sofa/component/io/mesh/MeshSTLLoader.h]:
```suggestion
    Data <unsigned int> _headerSize;
```

File: Sofa/Component/IO/Mesh/src/sofa/component/io/mesh/MeshSTLLoader.cpp [hunk 3]
```diff
@@ -43,10 +43,13 @@ static int MeshSTLLoaderClass = core::RegisterObject("Loader for the STL file fo
 
 //Base VTK Loader
 MeshSTLLoader::MeshSTLLoader() : MeshLoader()
-    , _headerSize(initData(&_headerSize, 80u, "headerSize","Size of the header binary file (just before the number of facet)."))
-    , _forceBinary(initData(&_forceBinary, false, "forceBinary","Force reading in binary mode. Even in first keyword of the file is solid."))
+    , d_headerSize(initData(&d_headerSize, 80u, "d_headerSize", "Size of the header binary file (just before the number of facet)."))
```
Comment (bakpaul) [inline Sofa/Component/IO/Mesh/src/sofa/component/io/mesh/MeshSTLLoader.cpp]:
```suggestion
    , d_headerSize(initData(&d_headerSize, 80u, "headerSize", "Size of the header binary file (just before the number of facet)."))
```

File: Sofa/Component/IO/Mesh/src/sofa/component/io/mesh/MeshTrianLoader.cpp [hunk 4]
```diff
@@ -39,14 +39,20 @@ int MeshTrianLoaderClass = core::RegisterObject("Specific mesh loader for trian
         ;
 
 MeshTrianLoader::MeshTrianLoader() : MeshLoader()
-    , p_trian2(initData(&p_trian2,(bool)false,"trian2","Set to true if the mesh is a trian2 format."))
-    , neighborTable(initData(&neighborTable,"neighborTable","Table of neighborhood triangle indices for each triangle."))
-    , edgesOnBorder(initData(&edgesOnBorder,"edgesOnBorder","List of edges which are on the border of the mesh loaded."))
-    , trianglesOnBorderList(initData(&trianglesOnBorderList,"trianglesOnBorderList","List of triangle indices which are on the border of the mesh loaded."))
+    , d_trian2(initData(&d_trian2, (bool)false, "trian2", "Set to true if the mesh is a trian2 format."))
+    , d_neighborTable(initData(&d_neighborTable, "d_neighborTable", "Table of neighborhood triangle indices for each triangle."))
```
Comment (bakpaul) [inline Sofa/Component/IO/Mesh/src/sofa/component/io/mesh/MeshTrianLoader.cpp]:
```suggestion
    , d_neighborTable(initData(&d_neighborTable, "neighborTable", "Table of neighborhood triangle indices for each triangle."))
```

File: Sofa/Component/IO/Mesh/src/sofa/component/io/mesh/MeshTrianLoader.cpp [hunk 5]
```diff
@@ -39,14 +39,20 @@ int MeshTrianLoaderClass = core::RegisterObject("Specific mesh loader for trian
...
+    , d_edgesOnBorder(initData(&d_edgesOnBorder, "d_edgesOnBorder", "List of edges which are on the border of the mesh loaded."))
```
Comment (bakpaul) [inline Sofa/Component/IO/Mesh/src/sofa/component/io/mesh/MeshTrianLoader.cpp]:
```suggestion
    , d_edgesOnBorder(initData(&d_edgesOnBorder, "edgesOnBorder", "List of edges which are on the border of the mesh loaded."))
```

File: Sofa/Component/IO/Mesh/src/sofa/component/io/mesh/MeshTrianLoader.cpp [hunk 6]
```diff
@@ -39,14 +39,20 @@ int MeshTrianLoaderClass = core::RegisterObject("Specific mesh loader for trian
...
+    , d_trianglesOnBorderList(initData(&d_trianglesOnBorderList, "d_trianglesOnBorderList", "List of triangle indices which are on the border of the mesh loaded."))
```
Comment (bakpaul) [inline Sofa/Component/IO/Mesh/src/sofa/component/io/mesh/MeshTrianLoader.cpp]:
```suggestion
    , d_trianglesOnBorderList(initData(&d_trianglesOnBorderList, "trianglesOnBorderList", "List of triangle indices which are on the border of the mesh loaded."))
```

File: Sofa/Component/IO/Mesh/src/sofa/component/io/mesh/MeshTrianLoader.h [hunk 7]
```diff
@@ -53,10 +53,22 @@ class SOFA_COMPONENT_IO_MESH_API MeshTrianLoader : public sofa::core::loader::Me
 
 public:
     //Add specific Data here:
-    Data <bool> p_trian2; ///< Set to true if the mesh is a trian2 format.
-    Data <type::vector< type::fixed_array <int,3> > > neighborTable; ///< Table of neighborhood triangle indices for each triangle.
-    Data <type::vector< type::vector<unsigned int> > > edgesOnBorder; ///< List of edges which are on the border of the mesh loaded.
-    Data <type::vector<unsigned int> > trianglesOnBorderList; ///< List of triangle indices which are on the border of the mesh loaded.
+    SOFA_ATTRIBUTE_DEPRECATED__RENAME_DATA()
+    Data <bool> trian2;
```
Comment (bakpaul) [inline Sofa/Component/IO/Mesh/src/sofa/component/io/mesh/MeshTrianLoader.h]:
```suggestion
    Data <bool> p_trian2;
```

File: Sofa/Component/IO/Mesh/src/sofa/component/io/mesh/MeshTrianLoader.cpp [hunk 8]
```diff
@@ -39,14 +39,20 @@ int MeshTrianLoaderClass = core::RegisterObject("Specific mesh loader for trian
...
 {
-    neighborTable.setPersistent(false);
-    edgesOnBorder.setPersistent(false);
-    trianglesOnBorderList.setPersistent(false);
+    d_neighborTable.setPersistent(false);
+    d_edgesOnBorder.setPersistent(false);
+    d_trianglesOnBorderList.setPersistent(false);
+
+    trian2.setParent(&d_trian2);
```
Comment (bakpaul) [inline Sofa/Component/IO/Mesh/src/sofa/component/io/mesh/MeshTrianLoader.cpp]:
```suggestion
    p_trian2.setParent(&d_trian2);
```

File: Sofa/Component/IO/Mesh/src/sofa/component/io/mesh/OffSequenceLoader.cpp [hunk 9]
```diff
@@ -39,8 +39,8 @@ int OffSequenceLoaderClass = core::RegisterObject("Read and load an .off file at
 
 OffSequenceLoader::OffSequenceLoader()
     : MeshOffLoader()
-    , nbFiles( initData(&nbFiles,(int)1,"nbOfFiles","number of files in the sequence") )
-    , stepDuration( initData(&stepDuration,0.04,"stepDuration","how long each file is loaded") )
+    , d_nbFiles(initData(&d_nbFiles, (int)1, "nbOfFiles", "number of files in the sequence") )
+    , d_stepDuration(initData(&d_stepDuration, 0.04, "d_stepDuration", "how long each file is loaded") )
```
Comment (bakpaul) [inline Sofa/Component/IO/Mesh/src/sofa/component/io/mesh/OffSequenceLoader.cpp]:
```suggestion
    , d_stepDuration(initData(&d_stepDuration, 0.04, "stepDuration", "how long each file is loaded") )
```

File: Sofa/Component/IO/Mesh/src/sofa/component/io/mesh/MeshSTLLoader.cpp (hunk 3 above)
Comment (alxbilger) [inline Sofa/Component/IO/Mesh/src/sofa/component/io/mesh/MeshSTLLoader.cpp]:
```suggestion
    , d_headerSize(initData(&d_headerSize, 80u, "headerSize", "Size of the header binary file (just before the number of facet)."))
```
Be careful in all of your PRs that you don't change the name of the Data.

File: Sofa/Component/IO/Mesh/src/sofa/component/io/mesh/MeshTrianLoader.cpp (hunk 6 above)
Comment (alxbilger) [inline Sofa/Component/IO/Mesh/src/sofa/component/io/mesh/MeshTrianLoader.cpp]:
```suggestion
    , d_neighborTable(initData(&d_neighborTable, "neighborTable", "Table of neighborhood triangle indices for each triangle."))
    , d_edgesOnBorder(initData(&d_edgesOnBorder, "edgesOnBorder", "List of edges which are on the border of the mesh loaded."))
    , d_trianglesOnBorderList(initData(&d_trianglesOnBorderList, "trianglesOnBorderList", "List of triangle indices which are on the border of the mesh loaded."))
```

File: Sofa/Component/IO/Mesh/src/sofa/component/io/mesh/OffSequenceLoader.cpp (hunk 9 above)
Comment (alxbilger) [inline Sofa/Component/IO/Mesh/src/sofa/component/io/mesh/OffSequenceLoader.cpp]:
```suggestion
    , d_stepDuration(initData(&d_stepDuration, 0.04, "stepDuration", "how long each file is loaded") )
```

File: Sofa/Component/Topology/Container/Grid/src/sofa/component/topology/container/grid/SparseGridTopology.cpp [hunk 10]
```diff
@@ -84,7 +84,7 @@ SparseGridTopology::SparseGridTopology(bool _isVirtual)
     , _max(initData(&_max, type::Vec3(0_sreal,0_sreal,0_sreal), "max","Max"))
     , _cellWidth(initData(&_cellWidth, (SReal)0.0, "cellWidth","if > 0 : dimension of each cell in the created grid"))
     , _nbVirtualFinerLevels( initData(&_nbVirtualFinerLevels, 0, "nbVirtualFinerLevels", "create virtual (not in the animation tree) finer sparse grids in order to dispose of finest information (usefull to compute better mechanical properties for example)"))
-    , dataResolution(initData(&dataResolution, type::Vec3i(0,0,0), "dataResolution", "Dimension of the voxel File"))
+    , dataResolution(initData(&dataResolution, type::Vec3i(0,0,0), "d_dataResolution", "Dimension of the voxel File"))
```
Comment (alxbilger) [inline Sofa/Component/Topology/Container/Grid/src/sofa/component/topology/container/grid/SparseGridTopology.cpp]:
```suggestion
    , dataResolution(initData(&dataResolution, type::Vec3i(0,0,0), "dataResolution", "Dimension of the voxel File"))
```

File: Sofa/Component/IO/Mesh/src/sofa/component/io/mesh/VTKExporter.h [hunk 11]
```diff
@@ -54,20 +54,65 @@ class SOFA_COMPONENT_IO_MESH_API VTKExporter : public core::objectmodel::BaseObj
...
-    Data<bool> writeTetras; ///< write tetra topology
-    Data<bool> writeHexas; ///< write hexa topology
-    Data<type::vector<std::string> > dPointsDataFields; ///< Data to visualize (on points)
-    Data<type::vector<std::string> > dCellsDataFields; ///< Data to visualize (on cells)
-    Data<unsigned int> exportEveryNbSteps; ///< export file only at specified number of steps (0=disable)
-    Data<bool> exportAtBegin; ///< export file at the initialization
-    Data<bool> exportAtEnd; ///< export file when the simulation is finished
-    Data<bool> overwrite; ///< overwrite the file, otherwise create a new file at each export, with suffix in the filename
+    SOFA_ATTRIBUTE_DEPRECATED__RENAME_DATA()
+    Data< std::string > vtkFilename;
```
Comment (bakpaul) [inline Sofa/Component/IO/Mesh/src/sofa/component/io/mesh/VTKExporter.h]:
```suggestion
    [Data< std::string >](sofa::core::objectmodel::DataFileName) vtkFilename;
```

**Review by alxbilger — CHANGES_REQUESTED** [review]

File: Sofa/Component/IO/Mesh/src/sofa/component/io/mesh/VoxelGridLoader.cpp [hunk 12]
```diff
@@ -41,20 +41,31 @@ int VoxelGridLoaderClass = RegisterObject("Voxel loader based on RAW files").add
...
-      idxInRegularGrid(initData(&idxInRegularGrid,"idxInRegularGrid","indices of the hexa in the grid.")),
-      backgroundValue ( initData ( &backgroundValue, "bgValue", "Background values (to be ignored)" ) ),
-      activeValue ( initData ( &activeValue, "dataValue", "Active data values" ) ),
-      generateHexa( initData ( &generateHexa, true, "generateHexa", "Interpret voxel as either hexa or points")),
+      d_voxelSize (initData (&d_voxelSize, Vec3 (1.0f, 1.0f, 1.0f ), "voxelSize", "Dimension of one voxel" ) ),
+      d_dataResolution (initData (&d_dataResolution, Vec3i (0, 0, 0 ), "resolution", "Resolution of the voxel file" ) ),
+      d_roi (initData (&d_roi, Vec6i (0, 0, 0, 0xFFFF, 0xFFFF, 0xFFFF ), "ROI", "Region of interest (xmin, ymin, zmin, xmax, ymax, zmax)" ) ),
+      d_headerSize (initData (&d_headerSize, 0, "header", "Header size in bytes" ) ),
+      d_segmentationHeaderSize (initData (&d_segmentationHeaderSize, 0, "segmentationHeader", "Header size in bytes" ) ),
+      d_idxInRegularGrid(initData(&d_idxInRegularGrid, "d_idxInRegularGrid", "indices of the hexa in the grid.")),
```
Comment (bakpaul) [inline Sofa/Component/IO/Mesh/src/sofa/component/io/mesh/VoxelGridLoader.cpp]:
```suggestion
      d_idxInRegularGrid(initData(&d_idxInRegularGrid, "idxInRegularGrid", "indices of the hexa in the grid.")),
```

File: Sofa/Component/IO/Mesh/src/sofa/component/io/mesh/VoxelGridLoader.cpp [hunk 13]
```diff
@@ -41,20 +41,31 @@ int VoxelGridLoaderClass = RegisterObject("Voxel loader based on RAW files").add
...
+      d_backgroundValue (initData (&d_backgroundValue, "bgValue", "Background values (to be ignored)" ) ),
+      d_activeValue (initData (&d_activeValue, "dataValue", "Active data values" ) ),
+      d_generateHexa(initData (&d_generateHexa, true, "d_generateHexa", "Interpret voxel as either hexa or points")),
```
Comment (bakpaul) [inline Sofa/Component/IO/Mesh/src/sofa/component/io/mesh/VoxelGridLoader.cpp]:
```suggestion
      d_generateHexa(initData (&d_generateHexa, true, "generateHexa", "Interpret voxel as either hexa or points")),
```

File: Sofa/Component/Mapping/Linear/src/sofa/component/mapping/linear/LineSetSkinningMapping.inl [hunk 14]
```diff
@@ -137,11 +137,12 @@ void LineSetSkinningMapping<TIn, TOut>::init()
...
                         lines[i].lineIndex = lines[i-1].lineIndex;
                         lines[i].weight = lines[i-1].weight;
-                        lines[i].position = lines[i-1].position;
+                        lines[i].position = lines[i - 1].position;
                     }
                     lines[lineInfluencedIndex].lineIndex = lineIndex;
                     lines[lineInfluencedIndex].weight = _weight;
-                    lines[lineInfluencedIndex].position = xfrom[line[0]].getOrientation().inverseRotate(xto[verticeIndex] - xfrom[line[0]].getCenter());
+                    lines[lineInfluencedIndex].
+                    position = xfrom[line[0]].getOrientation().inverseRotate(xto[verticeIndex] - xfrom[line[0]].getCenter());
```
Comment (bakpaul) [inline Sofa/Component/Mapping/Linear/src/sofa/component/mapping/linear/LineSetSkinningMapping.inl]: Add the break after the '=' instead...

File: Sofa/Component/Topology/Container/Grid/src/sofa/component/topology/container/grid/SparseGridTopology.cpp (hunk 10 above)
Comment (bakpaul) [inline Sofa/Component/Topology/Container/Grid/src/sofa/component/topology/container/grid/SparseGridTopology.cpp]:
```suggestion
    , dataResolution(initData(&dataResolution, type::Vec3i(0,0,0), "dataResolution", "Dimension of the voxel File"))
```

File: Sofa/Component/Topology/Container/Dynamic/src/sofa/component/topology/container/dynamic/MultilevelHexahedronSetTopologyContainer.cpp [hunk 15]
```diff
@@ -45,7 +45,7 @@ MultilevelHexahedronSetTopologyContainer::MultilevelHexahedronSetTopologyContain
     : HexahedronSetTopologyContainer(),
       _level(initData(&_level, 0, "level", "Number of resolution levels between the fine and coarse mesh")),
       fineResolution(initData(&fineResolution,Vec3i(0,0,0),"resolution","fine resolution")),
-      hexaIndexInRegularGrid(initData(&hexaIndexInRegularGrid,"idxInRegularGrid","indices of the hexa in the grid.")),
+      hexaIndexInRegularGrid(initData(&hexaIndexInRegularGrid,"d_idxInRegularGrid","indices of the hexa in the grid.")),
```
Comment (bakpaul) [inline Sofa/Component/Topology/Container/Dynamic/src/sofa/component/topology/container/dynamic/MultilevelHexahedronSetTopologyContainer.cpp]:
```suggestion
      hexaIndexInRegularGrid(initData(&hexaIndexInRegularGrid,"idxInRegularGrid","indices of the hexa in the grid.")),
```

File: Sofa/Component/Topology/Container/Dynamic/src/sofa/component/topology/container/dynamic/DynamicSparseGridTopologyContainer.cpp [hunk 16]
```diff
@@ -43,7 +43,7 @@ DynamicSparseGridTopologyContainer::DynamicSparseGridTopologyContainer()
     , resolution ( initData ( &resolution, type::Vec3i ( 0, 0, 0 ), "resolution", "voxel grid resolution" ) )
     , valuesIndexedInRegularGrid( initData ( &valuesIndexedInRegularGrid, sofa::type::vector<unsigned char>(), "valuesIndexedInRegularGrid", "values indexed in the Regular Grid" ) )
     , valuesIndexedInTopology( initData(&valuesIndexedInTopology, "valuesIndexedInTopology", "values indexed in the topology"))
-    , idxInRegularGrid( initData ( &idxInRegularGrid, sofa::type::vector<BaseMeshTopology::HexaID>(), "idxInRegularGrid", "indices in the Regular Grid" ) )
+    , idxInRegularGrid( initData ( &idxInRegularGrid, sofa::type::vector<BaseMeshTopology::HexaID>(), "d_idxInRegularGrid", "indices in the Regular Grid" ) )
```
Comment (bakpaul) [inline Sofa/Component/Topology/Container/Dynamic/src/sofa/component/topology/container/dynamic/DynamicSparseGridTopologyContainer.cpp]:
```suggestion
    , idxInRegularGrid( initData ( &idxInRegularGrid, sofa::type::vector<BaseMeshTopology::HexaID>(), "idxInRegularGrid", "indices in the Regular Grid" ) )
```

**Review by bakpaul — CHANGES_REQUESTED** [review]
Please be careful when you use 'search and replace' you don't want to rename the data name in the factory

**Comment by bakpaul** [comment]
I've put this PR in wip to avoid overloading the CI while you integrate all the comments. Once you've finished all the review, you can change the tag back to to review and then trigger the CI with the usual comment.

**Comment by lamriaimen** [comment]
[ci-build][with-all-tests]

**Comment by lamriaimen** [comment]
[ci-build][with-all-tests]

File: Sofa/framework/Helper/src/sofa/helper/MarchingCubeUtility.cpp [hunk 17]
```diff
@@ -633,7 +633,7 @@ namespace sofa::helper
                                     type::vector< type::vector<unsigned int> >*triangleIndexInRegularGrid,
                                     bool propagate ) const
     {
-        //    Vec3i gridSize = Vec3i ( dataResolution[0]/cubeStep, dataResolution[1]/cubeStep, dataResolution[2]/cubeStep );
+        //    Vec3i gridSize = Vec3i ( d_dataResolution[0]/cubeStep, d_dataResolution[1]/cubeStep, d_dataResolution[2]/cubeStep );
```
Comment (alxbilger) [inline Sofa/framework/Helper/src/sofa/helper/MarchingCubeUtility.cpp]: Not sure this file is intended to be modified

File: Sofa/framework/Helper/src/sofa/helper/MarchingCubeUtility.h [hunk 18]
```diff
@@ -220,7 +220,7 @@ class SOFA_HELPER_API MarchingCubeUtility
     type::Vec3i     dataResolution;
     type::Vec3     dataVoxelSize;
     BoundingBox bbox; //bbox used to remesh
-    BoundingBox roi; // Set value to 0 on this limit to always obtain manifold mesh. (Set to dataResolution by default but can be changed for ROI)
+    BoundingBox roi; // Set value to 0 on this limit to always obtain manifold mesh. (Set to d_dataResolution by default but can be changed for ROI)
```
Comment (alxbilger) [inline Sofa/framework/Helper/src/sofa/helper/MarchingCubeUtility.h]: Not sure this file is intended to be modified

**Review by alxbilger — CHANGES_REQUESTED** [review]

**Review by alxbilger — APPROVED** [review]

**Review by fredroy — APPROVED** [review]

**Pull request merged**